tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/ModuleKind.scala

```diff
@@ -0,0 +1,34 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js tools             **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2016, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package org.scalajs.core.tools.linker.backend
+
+/** Kind of module structure emitted for the Scala.js output. */
+sealed abstract class ModuleKind
+
+object ModuleKind {
+
+  /** No module structure.
+   *
+   *  With this module kind, exports are stored on the global object by
+   *  default, or to a separate object specified with
+   *  `__ScalaJSEnv.exportsNamespace`.
+   *
+   *  Imports are not supported.
+   */
+  case object NoModule extends ModuleKind
+
+  /** A Node.js module.
+   *
+   *  Imported modules are fetched with `require`. Exports go to the `exports`
+   *  module-global variable.
```

Comment:
Actually, no. The rationale in my head was that, if you were using this, it was because you want to use your module with Node.js. But that's not necessarily the case. One could generate CommonJS modules to be consumed by a later bundler like Webpack. I guess I implicitly discarded that scenario as, for such a use case, one would typically generate an ES6 module. But there could be people who already have an established CommonJS-based bundling pipeline, and don't want to migrate to something that understand ES6 modules.